geostore/check_files_checksums/utils.py

```diff
@@ -60,28 +55,44 @@ def validate(self, hash_key: str, range_key: str) -> None:
             )
             raise
 
-        try:
-            file_in_staging = self.validate_url_multihash(item.url, item.multihash)
-        except ChecksumMismatchError as error:
+        s3_response = self.get_s3_object(processing_item.url)
+
+        self.validate_url_multihash(
+            processing_item.url, processing_item.multihash, s3_response.response
+        )
+
+        processing_item.update(
+            actions=[
+                self.processing_assets_model.exists_in_staging.set(s3_response.file_in_staging)
+            ]
+        )
+
+    def validate_url_multihash(
+        self, url: str, hex_multihash: str, s3_file_object: StreamingBody
+    ) -> None:
+        checksum_function_code = int(hex_multihash[:2], 16)
+        checksum_function = FUNCS[checksum_function_code]
+
+        file_digest = checksum_function()
+        for chunk in s3_file_object.iter_chunks(chunk_size=CHUNK_SIZE):
+            file_digest.update(chunk)
+
+        if file_digest.digest() != decode(bytes.fromhex(hex_multihash)):
             content = {
-                MESSAGE_KEY: f"Checksum mismatch: expected {item.multihash[4:]},"
-                f" got {error.actual_hex_digest}"
+                MESSAGE_KEY: f"Checksum mismatch: expected {hex_multihash[4:]},"
+                f" got {file_digest.hexdigest()}"
             }
             self.log_failure(content)
             self.validation_result_factory.save(
-                item.url, Check.CHECKSUM, ValidationResult.FAILED, details=content
+                url, Check.CHECKSUM, ValidationResult.FAILED, details=content
             )
         else:
             self.logger.info(LOG_MESSAGE_VALIDATION_COMPLETE, extra={"outcome": Outcome.PASSED})
-            self.validation_result_factory.save(item.url, Check.CHECKSUM, ValidationResult.PASSED)
-            item.update(
-                actions=[self.processing_assets_model.exists_in_staging.set(file_in_staging)]
-            )
-
-    def validate_url_multihash(self, url: str, hex_multihash: str) -> bool:
+            self.validation_result_factory.save(url, Check.CHECKSUM, ValidationResult.PASSED)
 
+    def get_s3_object(self, url: str) -> GeostoreS3Response:
```

Comment:
I like this change - everything related to just getting the object in one place.